src/core/messages.ts

```diff
@@ -277,6 +293,130 @@ export interface Messages extends EmitsDiscontinuities {
   get channel(): Ably.RealtimeChannel;
 }
 
+interface MessageReactionEvent {
+  type: 'message-reaction-create' | 'message-reaction-remove';
+  refSerial: string;
+  refType: string;
+  reaction: string;
+  clientId: string;
+  timestamp: Date;
+}
+
+export type MessageReactionListener = (event: MessageReactionSummaryEvent) => void;
+export type MessageRawReactionListener = (event: MessageReactionEvent) => void;
+
+export interface Unsubscribable {
+  unsubscribe: () => void;
+}
+
+interface Reactions {
+  add(message: { serial: string }, refType: ReactionRefType, reaction: string, count?: number): Promise<void>;
+  remove(message: { serial: string }, refType: ReactionRefType, reaction: string): Promise<void>;
+  subscribe(listener: MessageReactionListener): Unsubscribable;
+  subscribeRaw(listener: MessageRawReactionListener): Unsubscribable;
+}
+
+export class DefaultMessageReactions implements Reactions {
+  private _emitter = new EventEmitter<{ reaction: MessageReactionEvent; summary: MessageReactionSummaryEvent }>();
+
+  constructor(
+    private readonly _logger: Logger,
+    private readonly _options: MessageOptions | undefined,
+    private readonly _api: ChatApi,
+    private readonly _roomID: string,
+    private readonly _channel: Ably.RealtimeChannel,
+  ) {
+    void _channel.subscribe(this._processMessageEvent.bind(this));
+    if (this._options?.RawMessageReactions) {
+      void _channel.annotations.subscribe(this._processAnnotationEvent.bind(this));
+    }
+  }
+
+  private _processAnnotationEvent(event: Ably.Annotation) {
+    console.log('ably-chat-js: annotation received:', event);
+  }
+
+  private _processMessageEvent(event: Ably.InboundMessage) {
+    // only process summary events
+    if (event.action !== ChatMessageActions.MessageAnnotationSummary) {
+      return;
+    }
+    // they must have a refSerial
+    if (!event.refSerial) {
+      return;
+    }
+    // they must have a version
+    if (!event.version) {
+      return;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/prefer-optional-chain, @typescript-eslint/no-unsafe-assignment
+    const single: Record<string, SingleReactionSummary> = (event.summary || {})[ReactionRefType.Single] || {};
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/prefer-optional-chain, @typescript-eslint/no-unsafe-assignment
+    const distinct: Record<string, DistinctReactionSummary> = (event.summary || {})[ReactionRefType.Distinct] || {};
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/prefer-optional-chain, @typescript-eslint/no-unsafe-assignment
+    const counter: Record<string, CounterReactionSummary> = (event.summary || {})[ReactionRefType.Counter] || {};
+
+    this._emitter.emit('summary', {
+      type: 'message-reaction-summary',
```

Comment:
Should we make this an enum? Its less error prone for users

Reply:
Done